hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/format/TestInputFormat.java

```diff
@@ -481,7 +489,7 @@ void testReadMORWithCompactionPlanScheduled() throws Exception {
       List<RowData> dataset = TestData.dataSetInsert(i + 1, i + 2);
       TestData.writeData(dataset, conf);
     }
-
+    createTableSource();
```

Comment:
Can we explain a little more why this test has been affected by your change ?

Reply:
If getTableSource method  is called before data is written, metaClient will be  null, and a call to tablesource.reset () in the unit test method  generate null pointer

Reply:
Thanks, i think the change is risky, because the null value is only handled in `HoodieTableSource#getStreamInputFormat`, original i put a reader meta client for the case where streaming reader starts but there is no data for the table (which is a common case for streaming read), so let's limit the change only to streaming read too and then i think we do not need to change the test case.

Reply:
@danny0405  ok,You mean change the logic below?   This will not affect batch reading：😄

<img width="927" alt="image" src="https://user-images.githubusercontent.com/104287562/169437477-bad7944e-0812-41c8-97e5-a4ee7b5249f0.png">

Reply:
ok,get it ! ~ 😄

Reply:
@danny0405    hi   danny,I have upgraded the program, please help me to  CR,thank you !~